src/features/broker-protection/actions/extract.js

```diff
@@ -205,33 +202,19 @@ function scrapedDataMatchesUserData (userData, scrapedData) {
         }
     }
 
-    if (scrapedData.addressCityState) {
-        // addressCityState is now being put in a list so can use matchAddressFromAddressListCityState
-        if (matchAddressFromAddressListCityState(userData.addresses, scrapedData.addressCityState)) {
-            matchedFields.push('addressCityState')
-            return { matchedFields, score: matchedFields.length, result: true }
-        }
-    }
-
-    // it's possible to have both addressCityState and addressCityStateList
-    if (scrapedData.addressCityStateList) {
-        if (matchAddressFromAddressListCityState(userData.addresses, scrapedData.addressCityStateList)) {
-            matchedFields.push('addressCityStateList')
-            return { matchedFields, score: matchedFields.length, result: true }
-        }
-    }
-
-    if (scrapedData.addressFull) {
-        if (matchesFullAddress(userData.addresses, scrapedData.addressFull)) {
-            matchedFields.push('addressFull')
-            return { matchedFields, score: matchedFields.length, result: true }
-        }
-    }
-
-    if (scrapedData.addressFullList) {
-        if (matchesFullAddressList(userData.addresses, scrapedData.addressFullList)) {
-            matchedFields.push('addressFullList')
-            return { matchedFields, score: matchedFields.length, result: true }
+    const addressFields = [
+        'addressCityState',
+        'addressCityStateList',
+        'addressFull',
+        'addressFullList'
+    ]
+
+    for (const addressField of addressFields) {
+        if (addressField in scrapedData) {
+            if (addressMatch(userData.addresses, scrapedData[addressField])) {
+                matchedFields.push(addressField)
+                return { matchedFields, score: matchedFields.length, result: true }
+            }
         }
     }
```

Comment:
because addresses are standardised everywhere now, via the address extractors, it means we can check them all for matches in the exact same way :)